src/GraphQL/Internal/AST.hs

```diff
@@ -45,24 +52,71 @@ module GraphQL.Internal.AST
 
 import Protolude hiding (Type)
 
-import Data.Char (isDigit)
+import qualified Data.Aeson as Aeson
 import qualified Data.Attoparsec.Text as A
-import GraphQL.Internal.Tokens (tok)
+import Data.Char (isDigit)
+import Test.QuickCheck (Arbitrary(..), elements, listOf)
 
--- TODO: Our GraphQL.Value.Name has more guarantees than AST.Name (which is
--- just a Text alias). We should change that so AST.Name is guaranteed valid.
+import GraphQL.Internal.Tokens (tok)
 
 -- * Name
 
-type Name = Text
+-- | A name in GraphQL.
+--
+-- https://facebook.github.io/graphql/#sec-Names
+newtype Name = Name { getNameText :: Text } deriving (Eq, Ord, Show)
```

Comment:
where is the `getNameText` style from? I find it a bit weird but can of course get used to it. I guess the main point is so people can write point-free instead of unpacking in the argument? I find the argument-unpacking more readable personally because it doesn't introduce a new identifier that I have to map back to the original type in my head, but that's very subjective.